src/rust/ide/src/executor/test_utils.rs

```diff
@@ -9,41 +9,50 @@ use crate::executor::global::spawn;
 
 #[derive(Debug)]
 pub struct TestWithLocalPoolExecutor {
-    executor    : executor::LocalPool,
-    is_finished : Rc<Cell<bool>>,
+    executor      : executor::LocalPool,
+    running_tasks : Rc<Cell<usize>>,
 }
 
 impl TestWithLocalPoolExecutor {
     pub fn set_up() -> Self {
-        let executor    = executor::LocalPool::new();
-        let is_finished = Rc::new(Cell::new(false));
+        let executor      = executor::LocalPool::new();
+        let running_tasks = Rc::new(Cell::new(0));
 
         set_spawner(executor.spawner());
-        Self {executor,is_finished}
+        Self {executor,running_tasks}
     }
 
-    pub fn run_test<Test>(&mut self, test:Test)
-    where Test : Future<Output=()> + 'static {
-        let is_finished_clone = self.is_finished.clone_ref();
+    pub fn run_task<Task>(&mut self, task: Task)
+    where Task : Future<Output=()> + 'static {
+        self.running_tasks.set(self.running_tasks.get() + 1);
+        let running_tasks_clone = self.running_tasks.clone_ref();
         spawn(async move {
-            test.await;
-            is_finished_clone.set(true);
+            task.await;
+            running_tasks_clone.set(running_tasks_clone.get() - 1);
         });
     }
 
     pub fn when_stalled<Callback>(&mut self, callback:Callback)
     where Callback : FnOnce() {
         self.executor.run_until_stalled();
-        if !self.is_finished.get() {
+        if self.running_tasks.get() > 0 {
             callback();
         }
     }
+
+    pub fn when_stalled_run_task<Task>(&mut self, task : Task)
```

Comment:
No, the case is a test `getting_module_during_load` in registry.rs. There I want to make sure, that first task will be polled first, then the second, because the second expects that load won't be called. If I would spawn two tasks at test beginning I have no guarantee which task will be run first.

Reply:
In essence, this is for defining order of _starting_ task, not finishing tasks. In other words, we want to run another task only when first task is during execution.